src/components/PlanRoom/RoomTables/Item.tsx

```diff
@@ -0,0 +1,184 @@
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { borderRadius },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  newTable = (id: number) => {
+    tables[id] = { top: 0, left: 0, text: id.toString(), width: 100, height: 50, borderRadius: 0 };
+    let top = Math.floor(Math.random() * 100) + 40;
+    if (top > 80) {
+      top = 80;
+    }
+    moveTable((id - 1).toString(), Math.floor(Math.random() * 800) + 50, top);
+  };
+
+  const [, drop] = useDrop(
+    () => ({
+      accept: ItemTypes.TABLE,
+      drop(item: TypeItem, monitor) {
+        const delta = monitor.getDifferenceFromInitialOffset() as XYCoord;
+        let left = item.left + delta.x;
+        let top = item.top + delta.y;
+        if (top > 319) {
+          const difference: number = top - 319;
+          top -= difference;
+        }
+        if (top < 0) {
+          top = 0;
+        }
+        if (left < 0) {
+          left = 0;
+        }
+        moveTable(item.id, left, top);
+        return undefined;
+      },
+    }),
+    [moveTable],
+  );
+
+  save = () => {
+    console.log(tables);
+    return tables;
+  };
+
+  return (
+    <Box ref={drop} w="100%" h="100%" position="relative">
+      {Object.keys(tables).map((id) => {
+        const { left, top, text, height, width, borderRadius } = tables[id];
+        return (
+          <Table
+            key={id}
+            id={id}
+            left={left}
+            top={top}
+            height={height}
+            width={width}
+            borderRadius={borderRadius}
+            text={text}
+          >
+            {text}
+          </Table>
+        );
+      })}
+    </Box>
+  );
+};
+
+const Table = ({ id, left, top, height, width, borderRadius, text }: TableType) => {
+  const [{ isDragging }, drag] = useDrag(
+    () => ({
+      type: ItemTypes.TABLE,
+      item: { id, left, top },
+      collect: (monitor) => ({
+        isDragging: monitor.isDragging(),
+      }),
+    }),
+    [id, left, top],
+  );
+
+  if (isDragging) {
+    return <Box ref={drag} />;
+  }
+  return (
+    <Textarea
+      id={id}
+      ref={drag}
+      position="absolute"
+      borderRadius={borderRadius}
+      cursor="pointer"
+      textAlign="center"
+      left={left}
+      top={top}
+      width={width}
+      height={height}
+      backgroundColor={backgroundColor}
+      color="white"
+      maxLength={2}
+      focusBorderColor="none"
+      paddingTop="25px"
+      maxH="80px"
+      defaultValue={text}
+      onDoubleClick={() => {
+        changeValueTableBorderRadius(id.toString(), TableRadius(borderRadius));
+      }}
+      onChange={(event) => {
+        changeValueTable(id.toString(), event.target.value.toString());
+      }}
+    />
+  );
+};
+
+const TableRadius = (radius: number) => {
+  if (radius == 40) {
+    return 0;
+  }
+  return 40;
+};
+
+let numberTable: number = 1;
+
+export const NewTable = () => {
```

Comment:
Functions should start with lowercase letter.

Reply:
Please apply it in all appropriate places so I don't have to write the same comment everywhere.

Reply:
Zastosowałem do najnowszej wersji sugestie Dominika : PascalCase